pkgs/tools/misc/memtest86-efi/default.nix

```diff
@@ -8,23 +8,13 @@
 
 stdenv.mkDerivation rec {
   pname = "memtest86-efi";
-  version = "8.4";
+  version = "9.3.1000";
 
   src = fetchzip {
-    # TODO: We're using the previous version of memtest86 because the
-    # company developing memtest86 has stopped providing a versioned download
-    # link for the latest version:
-    #
-    # https://www.passmark.com/forum/memtest86/44494-version-8-1-distribution-file-is-not-versioned?p=44505#post44505
-    #
-    # However, versioned links for the previous version are available, so that
-    # is what is being used.
-    #
-    # It does look like redistribution is okay, so if we had somewhere to host
-    # binaries that we make sure to version, then we could probably keep up
-    # with the latest versions released by the company.
-    url = "https://www.memtest86.com/downloads/memtest86-${version}-usb.zip";
-    sha256 = "sha256-jh4FKCYZbOQhRv6B7N8Hmw6RQCQvbBGaGFTMLwM1nk8=";
+    # We're using the Internet Archive Wayback Machine because the company developing MemTest86 has stopped providing a versioned download link for the latest version:
+    # https://forums.passmark.com/memtest86/44494-version-8-1-distribution-file-is-not-versioned
+    url = "https://web.archive.org/web/20211111004725/https://www.memtest86.com/downloads/memtest86-usb.zip";
```

Comment:
@aanderse Ah, sorry, I'm trying to say that since this binary is being downloaded from web.archive.org, you could potentially change the date in the URL to get an older version of the www.memtest86.com site, which would likely have `memtest86-efi-8.4` instead of 9.3.1000.

This would be a way to prevent bumping a major version on a stable branch.

-----------------------------------

Although like I said above, this would be slightly more work than just going ahead and merging this PR as-is.  I probably won't have time in the next week or two to review this, but I'm happy to hand this over to either of you, @aanderse @ajs124.  If you want to test and merge-in as-is, please feel free.  If you want to fix up the URL so that it gets the 8.4 version and test and merge-in, please feel free to do that instead.